pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -897,6 +903,40 @@ private static void validateFieldConfigList(@Nullable List<FieldConfig> fieldCon
     }
   }
 
+  /**
+   * Validates the compatibility of the indexes if the column has the forward index disabled. Throws exceptions due to
+   * compatibility mismatch. The checks performed are:
+   *     - Validate dictionary is enabled.
+   *     - Validate inverted index is enabled.
+   *     - Validate that either no range index exists for column or the range index version is at least 2 and isn't a
+   *       multi-value column (since mulit-value defaults to index v1).
+   */
+  private static void validateForwardIndexDisabledIndexCompatibility(String columnName, FieldConfig fieldConfig,
```

Comment:
(nit) we should consider keeping the check either here or during segment generation in `SegmentColumnarIndexCreator`. For now it is fine to safeguard multiple places until feature gets used for a while. 

I am guessing basic validation at the table config level will always be needed since it will come handy even during  enabling / disabling the feature on an existing column. So an unsupported combination will be caught when user attempts to update the table config in controller as opposed to detecting later on the reload path.

Reply:
Ok to keep for now but once we do the reload follow-ups, we should just keep this one because it is just duplicate code / harder to maintain

Reply:
yup agree. will definitely remove what we don't need for reload follow ups. Will also try this out in a perf cluster maybe to ensure that just the controller code path is sufficient to catch all cases.